### ISSUE TITLE
Decimal short formatting inconsistent with negative numbers

### ISSUE THREAD

**Issue by glennr** [elixir-cldr/cldr_numbers#30]
# Test

```
  test "Decimal currency short formatting" do
    assert MyApp.Cldr.Number.to_string(Decimal.new(1_000_000), format: :short) == {:ok, "1M"}
    assert MyApp.Cldr.Number.to_string(Decimal.new(-1_000_000), format: :short) == {:ok, "-1M"}
    assert MyApp.Cldr.Number.to_string(Decimal.new(100_000), format: :short) == {:ok, "100K"}
    assert MyApp.Cldr.Number.to_string(Decimal.new(-100_000), format: :short) == {:ok, "-100K"}
    assert MyApp.Cldr.Number.to_string(Decimal.new(1_000), format: :short) == {:ok, "1K"}
    assert MyApp.Cldr.Number.to_string(Decimal.new(-1_000), format: :short) == {:ok, "-1K"}
  end
```

# Expected

All tests pass

# Actual

All tests with negative numbers fail in similar ways:

```
  1) test Decimal currency short formatting (Number.Format.Test)
     test/number/number_format_test.exs:114
     Assertion with == failed
     code:  assert MyApp.Cldr.Number.to_string(Decimal.new(-1_000_000), format: :short) == {:ok, "-1M"}
     left:  {:ok, "-1,000,000"}
     right: {:ok, "-1M"}
     stacktrace:
       test/number/number_format_test.exs:118: (test)

  1) test Decimal currency short formatting (Number.Format.Test)
     test/number/number_format_test.exs:114
     Assertion with == failed
     code:  assert MyApp.Cldr.Number.to_string(Decimal.new(-100_000), format: :short) == {:ok, "-100K"}
     left:  {:ok, "-100,000"}
     right: {:ok, "-100K"}
     stacktrace:
       test/number/number_format_test.exs:120: (test)


  1) test Decimal currency short formatting (Number.Format.Test)
     test/number/number_format_test.exs:114
     Assertion with == failed
     code:  assert MyApp.Cldr.Number.to_string(Decimal.new(-1000), format: :short) == {:ok, "-1K"}
     left:  {:ok, "-1,000"}
     right: {:ok, "-1K"}
     stacktrace:
       test/number/number_format_test.exs:122: (test)
```
```

**Comment by kipcole9** [comment]
@glennr, thanks for the report and apologies for the inconvenience. Definitely a bug based upon these examples.  Will work on resolving this evening my time (UTC+8).

**Comment by glennr** [comment]
@kipcole9 hopefully the above MR saves you the trouble!

**Comment by kipcole9** [comment]
I was able to get the fix done more quickly than anticipated. I have published [ex_cldr_numbers version 2.27.1](https://hex.pm/packages/ex_cldr_numbers/2.27.1) with the following changelog:

### Bug Fixes

* Fix `:short` formatting of negative numbers. Thanks to @glennr for the report. Closes #30.

Thanks much for the detailed report and test cases, very helpful and much appreciated.

**Comment by glennr** [comment]
haha - you beat me to it.
